crates/swc_ecma_minifier/src/compress/pure/misc.rs

```diff
@@ -230,6 +230,49 @@ impl Pure<'_> {
         }
     }
 
+    /// new Array(...) -> Array(...)
+    pub(super) fn remove_new(&mut self, e: &mut Expr) {
+        if !self.options.pristine_globals {
+            return;
+        }
+
+        match e {
+            Expr::New(NewExpr {
+                span,
+                callee,
+                args,
+                type_args,
+            }) if callee.is_one_of_global_ref_to(
+                &self.expr_ctx,
+                &[
+                    // https://262.ecma-international.org/12.0/#sec-array-constructor
+                    "Array",
+                    // https://262.ecma-international.org/12.0/#sec-function-constructor
+                    "Function",
+                    // https://262.ecma-international.org/12.0/#sec-regexp-constructor
+                    "RegExp",
+                    // https://262.ecma-international.org/12.0/#sec-error-constructor
+                    "Error",
+                    // https://262.ecma-international.org/12.0/#sec-aggregate-error-constructor
+                    "AggregateError",
```

Comment:
Object too

Reply:
And [native errors](https://tc39.es/ecma262/#sec-native-error-types-used-in-this-standard)